src/coinbase/wallets/types.ts

```diff
@@ -0,0 +1,50 @@
+import type { Address } from "abitype";
+import type { LocalAccount } from "viem";
+import type { NetworkIdentifier } from "../../client";
+import {
+  SendUserOperationOptions,
+  SendUserOperationReturnType,
+} from "../actions/sendUserOperation";
+
+export type SmartWallet = {
+  address: Address;
+  account: LocalAccount;
+  networkId?: NetworkIdentifier;
+  type: "smart";
+  sendUserOperation: <T extends readonly unknown[]>(
+    options: SendUserOperationOptions<T>,
+  ) => Promise<SendUserOperationReturnType>;
+  useNetwork: (options: { networkId: NetworkIdentifier }) => void;
+};
+
+// In the future, we may introduce BaseWallet (or named differently to avoid Base chain confusion). BaseWallet would be a base class for all wallets.
+// We would have a PrivateKeyWallet type that extends BaseWallet and adds signTransaction which is specific to private key wallets.
+// We would have a HDKeyWallet type that extends the PrivateKeyWallet type and adds a method to get the HD key used by the private key wallet.
+// We would update the SmartWallet type to extend BaseWallet to add support for common wallet signing methods.
```

Comment:
This makes sense to me but I would definitely name it something other than Base wallet. I also think wallet isn't the most descriptive term for the functions this type encloses, `Signer` feels more accurate to me, then like `PrivateKeySigner`, `HDWalletSigner`. `SmartWallet` does make sense to me with type `<Signer & ...`